### PR TITLE
Implement GetThreadCount for AIX.

### PR DESCRIPTION
**Pull request by mtbrandy**

### CODE REVIEW

File: googletest/src/gtest-port.cc [hunk 1]
```diff
@@ -58,6 +58,10 @@
 # include <sys/procfs.h>
 #endif  // GTEST_OS_QNX
 
+#if GTEST_OS_AIX
+# include <procinfo.h>
```
Comment (BillyDonahue) [inline googletest/src/gtest-port.cc]: also #include <sys/types.h>, according to https://www-01.ibm.com/support/knowledgecenter/ssw_aix_71/com.ibm.aix.basetrf1/getprocs.htm

File: googletest/src/gtest-port.cc [hunk 2]
```diff
@@ -146,6 +150,19 @@ size_t GetThreadCount() {
   }
 }
 
+#elif GTEST_OS_AIX
+
+size_t GetThreadCount() {
+  struct procentry64 entry;
```
Comment (BillyDonahue) [inline googletest/src/gtest-port.cc]: AIX has getprocs and getprocs64, depending on whether you're building for a 32-bit or 64-bit target.
How do you know you're building specifically for 64-bit AIX?

Comment (mtbrandy) [inline googletest/src/gtest-port.cc]: getprocs64 can be used for either execution mode.  It simply provides a size-invariant procentry64 structure.

File: googletest/src/gtest-port.cc (hunk 1 above)
Comment (mtbrandy) [inline googletest/src/gtest-port.cc]: sys/types.h provides the definition for pid_t.  This is implicitly included in any of the std.* headers.  I will explicitly add it if you want.

File: googletest/src/gtest-port.cc (hunk 2 above)
Comment (BillyDonahue) [inline googletest/src/gtest-port.cc]: Ah yes. You're right: "Alternatively, the struct procsinfo64 and sizeof (struct procsinfo64) can be used by 32-bit getprocs to return full 64-bit process information"

File: googletest/src/gtest-port.cc (hunk 1 above)
Comment (BillyDonahue) [inline googletest/src/gtest-port.cc]: Please add it explicitly. I don't want to rely on the presence of other includes. Those could disappear.

Comment (mtbrandy) [inline googletest/src/gtest-port.cc]: Done.

**Comment by mtbrandy** [comment]
Anything else you need from me to close this out?

**Comment by mtbrandy** [comment]
Any update here?  The failing check appears to be unrelated.

**Comment by BillyDonahue** [comment]
Looks good to me

**Pull request merged**